### ISSUE TITLE
Library not loaded: @rpath/NodeMediaClient.framework/NodeMediaClient

### ISSUE THREAD

**Issue by aymather** [NodeMedia/react-native-nodemediaclient#71]
I am able to successfully `yarn install react-native-nodemediaclient` into my project and `react-native run-ios` start it with a simulator. But when I try to run it on a device by opening the [PROJECT NAME].xcworkspace with xcode -> selecting my connected iphone as the build target -> pressing run, I get the following error.

```
dyld: Library not loaded: @rpath/NodeMediaClient.framework/NodeMediaClient
  Referenced from: /private/var/containers/Bundle/Application/A46F7183-3C90-4CDA-B559-5BCC902B9626/udio.app/udio
  Reason: no suitable image found.  Did find:
	/private/var/containers/Bundle/Application/A46F7183-3C90-4CDA-B559-5BCC902B9626/udio.app/Frameworks/NodeMediaClient.framework/NodeMediaClient: code signature invalid for '/private/var/containers/Bundle/Application/A46F7183-3C90-4CDA-B559-5BCC902B9626/udio.app/Frameworks/NodeMediaClient.framework/NodeMediaClient'

	/private/var/containers/Bundle/Application/A46F7183-3C90-4CDA-B559-5BCC902B9626/udio.app/Frameworks/NodeMediaClient.framework/NodeMediaClient: code signature invalid for '/private/var/containers/Bundle/Application/A46F7183-3C90-4CDA-B559-5BCC902B9626/udio.app/Frameworks/NodeMediaClient.framework/NodeMediaClient'

	/private/var/containers/Bundle/Application/A46F7183-3C90-4CDA-B559-5BCC902B9626/udio.app/Frameworks/NodeMediaClient.framework/NodeMediaClient: stat() failed with errno=1
	/private/var/containers/Bundle/Application/A46F7183-3C90-4CDA-B559-5BCC902B9626/udio.app/Frameworks/NodeMediaClient.framework/NodeMediaClient: code signature invalid for '/private/var/containers/Bundle/Application/A46F7183-3C90-4CDA-B559-5BCC902B9626/udio.app/Frameworks/NodeMediaClient.framework/NodeMediaClient'

	/private/var/containers/Bundle/Application/A46F7183-3C90-4CDA-B559-5BCC902B9626/udio.app/Frameworks/NodeMediaClient.framework/NodeMediaClient: stat() failed with errno=1
(lldb) 
```

I tried downloading the example and running it but it also ran into errors even after a successful clone, `yarn install`, and `pod install`

Is there something I'm missing that you have to do within xcode for linking this as a dependency? There isn't much documentation :(

**Comment by graffme** [comment]
@aymather Did you solve this?

**Comment by aymather** [comment]
I did solve the issue @graffme, but then I ran into another issue, so I'll explain my answer below, just know that once I got it loaded, the NodeCameraView was not able to access my camera on both ios and android. And I haven't been able to solve that yet.

What you need to do is link the embedded binary (which is the .framework file) to your project. Here's how you do that:

1. You need to open the file [YOUR PROJECT NAME].xcworkspace in xcode
^ you can find this file in the path "/ios/[YOUR PROJECT NAME].xcworkspace"

2. Once you have that workspace opened, and XCode has finished processing the files, click on the folder icon in your toolbar which should open your project target, and the Pods directory like so... 
<img width="411" alt="Screen Shot 2020-04-10 at 10 15 05 AM" src="https://user-images.githubusercontent.com/41848756/79009448-4b3be400-7b14-11ea-9b8d-529480e92aa8.png">

3. Once you've selected your target, click on the Build Phases tab in the toolbar like so...
<img width="1134" alt="Screen Shot 2020-04-10 at 10 17 23 AM" src="https://user-images.githubusercontent.com/41848756/79009545-8e965280-7b14-11ea-8a6d-75a4f89c09f7.png">

4. Open "Link Binary With Libraries" and click on the little plus button to add a framework
<img width="965" alt="Screen Shot 2020-04-10 at 10 19 40 AM" src="https://user-images.githubusercontent.com/41848756/79009651-dc12bf80-7b14-11ea-8a7e-59026afc1653.png">

5. When you open the popup, click on the dropdown titled "Add Other" and select "Add Files"
<img width="408" alt="Screen Shot 2020-04-10 at 10 20 28 AM" src="https://user-images.githubusercontent.com/41848756/79009708-05cbe680-7b15-11ea-9496-eea5361d6f6a.png">

6. Now you need to find the NodeMediaClient.framework file included in the Pod package. Mine was located in (starting from the PROJECT root), "/ios/Pods/NodeMediaClient/NodeMediaClient/NodeMediaClient.framework". Select it, and open it.
<img width="793" alt="Screen Shot 2020-04-10 at 10 23 21 AM" src="https://user-images.githubusercontent.com/41848756/79009845-580d0780-7b15-11ea-9b38-702f9d1ca7de.png">

7. If you do not do this last step, it will not work. You need to change the "Required" option in the framework you just linked, to "Optional"
<img width="963" alt="Screen Shot 2020-04-10 at 10 24 32 AM" src="https://user-images.githubusercontent.com/41848756/79009888-7e32a780-7b15-11ea-865f-312f54914c28.png">

8. Now if you have a project structure like mine with the [Project Name]Tests target autogenerated for you, then you need to click on that target as well, and just repeat the same process of linking the binary.
<img width="161" alt="Screen Shot 2020-04-10 at 10 25 00 AM" src="https://user-images.githubusercontent.com/41848756/79009961-ac17ec00-7b15-11ea-95ef-0150424275b6.png">
So just click on that target ^ and rinse -> repeat

Let me know if something was unclear or if it doesn't work for you, happy to help :)

**Comment by graffme** [comment]
@aymather you're **the best man** with those screenshots! 🙏
The interesting thing is that I'm using NodeCameraView component and streaming video to mux.com. The component is trying to send asset (so it is working) I can see it on mux.com in events, but there is no video or audio. My latest guess is that maybe there is a problem with permissions and it has no access? Gonna dig into it 🔎What is this another issue you run into that you've mentioned?

**Comment by aymather** [comment]
No problem @graffme !

If the issue you're experiencing is with permissions, then I have the solution for that as well.

You need to add some code to the Info.plist file.

Go to "/ios/[PROJECT NAME]/Info.plist" and add the following code.
```
<key>NSPhotoLibraryUsageDescription</key>               <!-- added this for photo library permission -->
<string>Need library access to upload images</string> <!-- added this for photo library permission -->
<key>NSCameraUsageDescription</key> <!-- added this for camera permission -->
<string>Need camera access to take pictures</string> <!-- added this for camera permission -->
```

And the other issue that I ran into is just that the camera won't show up on my ios device. I was using mux.com too, and it wasn't even receiving the signal in mux, i.e. nothing happened in my mux dashboard when I hit `this.vb.start()`

How did you know that it was sending the signal? Just because mux.com was displaying something happening in the dashboard? Did you have to solve any problems to make this happen?

**Comment by aymather** [comment]
That code just needs to be a child of the <dict> tag by the way, if that wasn't clear.

**Comment by graffme** [comment]
Yeah I did added permissions in Info.plis for Camera and Microphone, but I think this might not be enough, cus (I'm not like 100% sure if this is a case in dev mode), but when I run the app for the first time it's not asking me for permissions (should prompt). 

In mux.com I created a live stream and used link rtmp://global-live.mux.com:5222/app/{stream-key}. Then in events I got such logs:
<img width="310" alt="Screenshot 2020-04-10 at 22 27 15" src="https://user-images.githubusercontent.com/20944432/79020968-9c7ebe80-7b7a-11ea-84f3-d8a54e790342.png"> 
And in the _video.assed.errored_ there is a message "The input stream did not contain audio or video." 

I'm using hooks so I'm just creating a "stream" ref with useRef() and call the component with stream.current.start()/stop()

**Comment by graffme** [comment]
```javascript
<NodeCameraView
        style={StyleSheet.absoluteFill}
        ref={stream}
        outputUrl={`rtmp://global-live.mux.com:5222/app/${streamKey}`}
        camera={streamSettings.camera}
        audio={streamSettings.audio}
        video={streamSettings.video}
        autopreview
/>
```

StreamSettings same as in README

**Comment by aymather** [comment]
I'm starting to think that this repo just isn't up to date. It hasn't had a commit in a while. I think something with React Native has changed and maybe this package just isn't accessing the camera the way it's supposed to. I downloaded react-native-camera and was able to connect to the camera just fine. It asked me for permissions and everything. But if you find the solution, please let me know :)

**Comment by graffme** [comment]
Yeah, I'm afraid this might be the case, but thank you a lot anyway!

**Comment by bfarrgaynor** [comment]
Have you guys found another solution to streaming to mux from react-native? This is a major blow to my project if I can't get streaming working in RN. Even another provider.

**Comment by aymather** [comment]
Trust me I was in the same position @bfarrgaynor . I finally gave up and switched to writing my app natively. I can point you in some good directions for Swift if you're doing iOS but I don't know android very well.

I honestly think there isn't a single working react-native package for streaming :(

**Comment by bfarrgaynor** [comment]
We are pretty married to RN here. I think I'm going to try WebRTC for this. I will post what I find out. Wish me luck.

**Comment by illuspas** [comment]
This package has been updated to 0.1.9.
The problem should have been fixed.
It worked fine on my iPhone 13.3.1，Android 10.
